controllers/addon_controller.go

```diff
@@ -123,20 +123,28 @@ func (r *AddonReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 		return reconcile.Result{}, ignoreNotFound(err)
 	}
+	
+	return r.execAddon(ctx, req, log, instance)
+}
 
+func (r *AddonReconciler) execAddon(ctx context.Context, req reconcile.Request, log logr.Logger, instance *addonmgrv1alpha1.Addon) (reconcile.Result, error) {
+	defer func() {
+		if err := recover(); err != nil {
+			log.Info("Error: Panic occurred during execAdd %s/%s due to %s", instance.Namespace, instance.Name, err)
+		}
+	}()
 	// Process addon instance
 	ret, err := r.processAddon(ctx, req, log, instance)
 
-	// Always update status, cache
+	// Always update cache, status
 	err = r.updateAddonStatus(ctx, instance)
 	if err != nil {
 		// Force retry when status fails to update
 		ret.Requeue = true
 		return ret, err
 	}
 	r.addAddonToCache(instance)
-
-	return ret, err
+	return ret, nil
```

Comment:
I think we found a logic error from past... if there's no panic but a legit error here we wanted to return the error from `processAddon`, can we change this return to `procErr` and above code to `procErr := r.processAddon()`?

Reply:
why err is nil?  I thought there will be err, right?

Reply:
agree. it makes more clear.